src/Components/test/testassets/Components.TestServer/RazorComponentEndpointsStartup.cs

```diff
@@ -86,7 +86,17 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
 
             _ = app.UseEndpoints(endpoints =>
             {
-                endpoints.MapStaticAssets();
+                // REVIEW: Why doesn't MapStaticAssets search env.ContentRootPath before AppContext.BaseDirectory first to begin with?
```

Comment:
@javiercn Would it make sense for `MapStaticAssets()` to search the content root first for the manifest by default? This would make things behave more similarly to `UseStaticFiles()` and probably prevent a lot of future bug reports.

It's designed to be the configurable root for things like static assets manifests

> The content root is the base path for:
>
> * The executable hosting the app (*.exe*).
> * Compiled assemblies that make up the app (*.dll*).
> * Content files used by the app, such as:
>  * Razor files (`.cshtml`, `.razor`)
>  * Configuration files (`.json`, `.xml`)
>  * Data files (`.db`)
> * The [Web root](#web-root), typically the *wwwroot* folder.

https://learn.microsoft.com/aspnet/core/fundamentals/#content-root

Of course, it makes sense to fall back to `AppContext.BaseDirectory` primarily for development where the content root is the project root which will not contain the manifest. But I think it makes sense to look in the user-configurable location first.

Reply:
@halter73 we use `AppContext.BaseDirectory` because that's where the system generates the manifest during build/publish. I believe that's the "correct" thing to do.

As for looking at the "ContentRoot" i'm not sure if it's correct. When will the file not be on the same location as the assembly?

Reply:
I thought it might be useful for people writing custom msbuild logic to publish wasm dependencies to a nonstandard PublishDir like our tests. Those people could make that work before with UseStaticFiles by modifying their content root like our tests did, and I was wondering if we should keep that working.

It's an advanced scenario though, and MapStaticAssets is a new thing. It's probably fine to say those people just need to specify the full path to the manifest.